internal/cli/trigger_list.go

```diff
@@ -0,0 +1,191 @@
+package cli
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/posener/complete"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+)
+
+type TriggerListCommand struct {
+	*baseCommand
+
+	flagTriggerLabels []string
+
+	flagJson bool
+}
+
+func (c *TriggerListCommand) Run(args []string) int {
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	flagSet := c.Flags()
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+	ctx := c.Ctx
+
+	req := &pb.ListTriggerRequest{
+		Labels: c.flagTriggerLabels,
+	}
+
+	if c.flagWorkspace != "" {
+		req.Workspace = &pb.Ref_Workspace{
+			Workspace: c.flagWorkspace,
+		}
+	}
+
+	if c.flagProject != "" {
+		req.Project = &pb.Ref_Project{
+			Project: c.flagProject,
+		}
+	}
+
+	if c.flagApp != "" {
+		req.Application = &pb.Ref_Application{
+			Project:     c.flagProject,
+			Application: c.flagApp,
+		}
+	}
+
+	resp, err := c.project.Client().ListTriggers(ctx, req)
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+
+	if len(resp.Triggers) == 0 {
+		return 0
+	}
+
+	if c.flagJson {
+		var m jsonpb.Marshaler
+		m.Indent = "\t"
+		for _, t := range resp.Triggers {
+			str, err := m.MarshalToString(t)
+			if err != nil {
+				c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+				return 1
+			}
+
+			fmt.Println(str)
+		}
+		return 0
+	}
+
+	c.ui.Output("Trigger URL Configs", terminal.WithHeaderStyle())
+
+	tbl := terminal.NewTable("ID", "Name", "Workspace", "Project", "Application", "Operation", "Description", "Labels", "Last Time Active")
```

Comment:
Looks good, but also so wide! I wonder if we should steal an idea from kubectl and have a short version of commands like this, with maybe only the 5 most important columns, and add a `-o wide` option to include all the columns. This is exacerbated by our existing line-wrapping problems: https://github.com/hashicorp/waypoint/issues/1608

Other commands are guilty too though, so it shouldn't block this PR. Just an idea.

Reply:
I could see that. I also wouldn't mind punting on it and we could fix it across all commands and use the same flag name 🤔 There's also always the `-json` flag which is nicely formatted and not as wide